src/cli/index.js

```diff
@@ -227,11 +235,25 @@ module.exports = function cli (params = {}, callback) {
       lastEvent = Date.now()
     })
 
-    /**
-     * Watch for sandbox errors
-     */
+    // Watch for sandbox errors
     watcher.on('error', function (err) {
       update.error(`Error:`, err)
     })
+
+    // Workaround for https://github.com/yuanchuan/node-watch/issues/105
+    // wait until watcher is ready before calling back, and when we do call
+    // back, provide a function to close the sandbox server, the watcher, and
+    // the stdin reader.
+    watcher.on('ready', function () {
+      if (callback) callback(null, function close () {
```

Comment:
I'm wary of calling back with a close function, we used to do this before and it was a frequent source of confusion – moreover, with the 2.0 refactor I really strove to give it a consistent API for starting and stopping everything.

Is it fair to say this is really only something we need for testing at this moment?

Reply:
Fair to say, yes. Previously we called back with nothing and this is what led to the situation of relying on `process.exit` in the tests 😒

Reply:
Is your hesitance with the fact that it's a function or that we are passing anything back at all?

For context, I toyed with passing back an object that had references to the sandbox server and the watcher, but then you'd need to remember to pause reading `stdin` yourself on top of closing the watcher and ending the sandbox server... figured wrapping that up in a function was easier.

Reply:
Update, looks like this may be fixed in `node-watch` land! See: https://github.com/yuanchuan/node-watch/issues/105#issuecomment-739458956

wdyt?

Reply:
I will most definitely give it a shot and see how it works!

However, I think the general pattern of passing some manner of function or object containing references to everything that needs to be closed/cleaned up is still necessary for anyone consuming sandbox as a module (in particular the requirement to call `process.stdin.pause()` to free up the process).

Reply:
Hey @ryanblock , I'm revisiting this. Is the function-invocation pattern still something you are set against? If so, I can still merge this PR I will just remove all the tests (since the cli module is basically untestable in a single process).